### PR TITLE
Split e2e-provisioning test into two steps

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Split e2e-provisioning test on 2 phases:
   - provisioning
   - cleanup

logs from provisioning:
```
Starting e2e-provisioning test on GCP. Waiting 20s for api server...
time="2020-04-08T08:04:03Z" level=info msg="Provisioning Runtime [instanceID: 199c898a-d2ab-4650-ad0e-6417654205ea, NAME: e2e-provisioning-lywfhfhcjr]" service=broker_client
time="2020-04-08T08:04:03Z" level=info msg="Provisioning parameters: {\"service_id\":\"47c9dcbf-ff30-448e-ab36-d3bad66ba281\",\"plan_id\":\"ca6e5357-707f-4565-bbbd-b3ab732597c6\",\"organization_guid\":\"3a6ca609-4903-45f5-9575-602b0662de5e\",\"space_guid\":\"b3417ee6-25fc-43fe-92cc-d6c9659f149a\",\"context\":{\"tenant_id\":\"1eba80dd-8ff6-54ee-be4d-77944d17b10b\",\"subaccount_id\":\"39ba9a66-2c1a-4fe4-a28e-6e5db434084e\",\"globalaccount_id\":\"3e64ebae-38b5-46a0-b1ed-9ccee153a0ae\"},\"parameters\":{\"name\":\"e2e-provisioning-lywfhfhcjr\",\"components\":[]},\"maintenance_info\":{\"version\":\"0.1.0\",\"description\":\"Kyma environment broker e2e-provisioning test\"}}" service=broker_client
time="2020-04-08T08:04:04Z" level=info msg="Successfully send provision request, got operation ID 85eb72bb-4240-409e-8d2e-7b44925bfe5c" service=broker_client
time="2020-04-08T08:04:04Z" level=info msg="Creating config map e2e-runtime-config-gcp with test data"
time="2020-04-08T08:04:04Z" level=info msg="Waiting for operation at most 3h0m0s" service=broker_client
time="2020-04-08T08:09:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:14:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:19:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:24:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:29:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Last operation status: succeeded" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Operation succeeded!" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Fetching the Runtime's dashboard URL" service=broker_client
time="2020-04-08T08:34:05Z" level=info msg="Successfully fetched dashboard URL: https://console.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com" service=broker_client
time="2020-04-08T08:34:05Z" level=info msg="Updating config map e2e-runtime-config-gcp with dashboardUrl"
time="2020-04-08T08:34:05Z" level=info msg="Fetching runtime's kubeconfig"
time="2020-04-08T08:34:05Z" level=info msg="Create request to director service" service=director_client
time="2020-04-08T08:34:05Z" level=info msg="Send request to director" service=director_client
time="2020-04-08T08:34:05Z" level=info msg="Getting authorization token for credentials to access Director from endpoint: https://oauth2.mps.dev.kyma.cloud.sap/oauth2/token"
time="2020-04-08T08:34:06Z" level=error msg="Successfully unmarshal response oauth token for accessing Director"
time="2020-04-08T08:34:06Z" level=info msg=">> variables: map[]"
time="2020-04-08T08:34:06Z" level=info msg=">> query: query {\n\tresult: runtimes(filter: { key: \"broker_instance_id\" query: \"\\\"199c898a-d2ab-4650-ad0e-6417654205ea\\\"\" }) {\n    data {\n      id\n\t}\n}\n}"
time="2020-04-08T08:34:06Z" level=info msg=">> headers: map[Accept:[application/json; charset=utf-8] Authorization:[Bearer bi_SsNePyleTRTZ5j9b_CST8t9apxjRwoXPO1ALM-kc.yYlwvaAm18iEGIaiq_rNbmzFLs0OX8gFaoeZVYBjIcQ] Content-Type:[application/json; charset=utf-8] Tenant:[3e64ebae-38b5-46a0-b1ed-9ccee153a0ae]]"
time="2020-04-08T08:34:06Z" level=info msg="<< {\"data\":{\"result\":{\"data\":[{\"id\":\"14644e6b-fbbb-4b3a-b8ac-aa97ce48f814\"}]}}}"
time="2020-04-08T08:34:06Z" level=info msg="Extract the RuntimeID from the response" service=director_client
time="2020-04-08T08:34:06Z" level=info msg="Creating a secret e2e-runtime-config-gcp with test data"
time="2020-04-08T08:34:06Z" level=info msg="Calling the dashboard URL: https://dex.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com/auth?client_id=console&nonce=79bf9c29&redirect_uri=https%3A%2F%2Fconsole.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com&response_type=id_token+token&scope=audience%3Aserver%3Aclient_id%3Akyma-client+audience%3Aserver%3Aclient_id%3Aconsole+openid+profile+email+groups&state=5a4f3d15" service=dashboard_checker
time="2020-04-08T08:34:06Z" level=info msg="Successful response from the dashboard URL" service=dashboard_checker
PASS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2  100     2    0     0   2000      0 --:--:-- --:--:-- --:--:--  2000
OK%
```

logs from cleanup:
```
Starting e2e-provisioning cleanup test on GCP. Waiting 20s for api server...
time="2020-04-08T09:58:45Z" level=info msg="using instance ID 9bfde7f7-8036-450d-b41e-0d812145177e"
time="2020-04-08T09:58:46Z" level=info msg="Cleaning up..."
time="2020-04-08T09:58:46Z" level=info msg="removing secret e2e-runtime-config-gcp"
time="2020-04-08T09:58:46Z" level=warning msg="secret not found"
time="2020-04-08T09:58:46Z" level=info msg="removing config map e2e-runtime-config-gcp"
time="2020-04-08T09:58:47Z" level=info msg="Create request to director service" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Send request to director" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Getting authorization token for credentials to access Director from endpoint: https://oauth2.mps.dev.kyma.cloud.sap/oauth2/token"
time="2020-04-08T09:58:47Z" level=error msg="Successfully unmarshal response oauth token for accessing Director"
time="2020-04-08T09:58:47Z" level=info msg=">> variables: map[]"
time="2020-04-08T09:58:47Z" level=info msg=">> query: query {\n\tresult: runtimes(filter: { key: \"broker_instance_id\" query: \"\\\"9bfde7f7-8036-450d-b41e-0d812145177e\\\"\" }) {\n    data {\n      id\n\t}\n}\n}"
time="2020-04-08T09:58:47Z" level=info msg=">> headers: map[Accept:[application/json; charset=utf-8] Authorization:[Bearer taXIhaNoqj8CkV14HrZpvOwKYNyvvaU9LtDNErthZWU.e0zX_sHVLS8XiBdFuu1HRSA4EjVlOZ6B2_Gdhj5mHyA] Content-Type:[application/json; charset=utf-8] Tenant:[3e64ebae-38b5-46a0-b1ed-9ccee153a0ae]]"
time="2020-04-08T09:58:47Z" level=info msg="<< {\"data\":{\"result\":{\"data\":[{\"id\":\"8bf252c1-8fa4-4836-8b9c-b11c1a00b9f9\"}]}}}"
time="2020-04-08T09:58:47Z" level=info msg="Extract the RuntimeID from the response" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Waiting for uaa-issuer instance to be removed" service=runtime_client
time="2020-04-08T09:58:48Z" level=info msg="Successfully ensured UAA instance was removed" service=runtime_client
time="2020-04-08T09:58:48Z" level=info msg="Deprovisioning Runtime [ID: 9bfde7f7-8036-450d-b41e-0d812145177e, NAME: e2e-provisioning-dtju64uqv6]" service=broker_client
time="2020-04-08T09:58:49Z" level=info msg="Successfully send deprovision request, got operation ID d852610f-857d-4f63-b41b-5255666cd78f" service=broker_client
time="2020-04-08T09:58:49Z" level=info msg="Waiting for operation at most 1h0m0s" service=broker_client
time="2020-04-08T10:03:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:08:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:13:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:18:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:23:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:28:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:33:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:38:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:43:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:48:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:53:49Z" level=info msg="Last operation status: succeeded" service=broker_client
time="2020-04-08T10:53:49Z" level=info msg="Operation succeeded!" service=broker_client
PASS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2  100     2    0     0   1000      0 --:--:-- --:--:-- --:--:--  1000
OK
```

**Related issue(s)**

### CODE REVIEW

**Comment by polskikiel** [comment]
logs from provisioning:
```
Starting e2e-provisioning test on GCP. Waiting 20s for api server...
time="2020-04-08T08:04:03Z" level=info msg="Provisioning Runtime [instanceID: 199c898a-d2ab-4650-ad0e-6417654205ea, NAME: e2e-provisioning-lywfhfhcjr]" service=broker_client
time="2020-04-08T08:04:03Z" level=info msg="Provisioning parameters: {\"service_id\":\"47c9dcbf-ff30-448e-ab36-d3bad66ba281\",\"plan_id\":\"ca6e5357-707f-4565-bbbd-b3ab732597c6\",\"organization_guid\":\"3a6ca609-4903-45f5-9575-602b0662de5e\",\"space_guid\":\"b3417ee6-25fc-43fe-92cc-d6c9659f149a\",\"context\":{\"tenant_id\":\"1eba80dd-8ff6-54ee-be4d-77944d17b10b\",\"subaccount_id\":\"39ba9a66-2c1a-4fe4-a28e-6e5db434084e\",\"globalaccount_id\":\"3e64ebae-38b5-46a0-b1ed-9ccee153a0ae\"},\"parameters\":{\"name\":\"e2e-provisioning-lywfhfhcjr\",\"components\":[]},\"maintenance_info\":{\"version\":\"0.1.0\",\"description\":\"Kyma environment broker e2e-provisioning test\"}}" service=broker_client
time="2020-04-08T08:04:04Z" level=info msg="Successfully send provision request, got operation ID 85eb72bb-4240-409e-8d2e-7b44925bfe5c" service=broker_client
time="2020-04-08T08:04:04Z" level=info msg="Creating config map e2e-runtime-config-gcp with test data"
time="2020-04-08T08:04:04Z" level=info msg="Waiting for operation at most 3h0m0s" service=broker_client
time="2020-04-08T08:09:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:14:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:19:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:24:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:29:04Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Last operation status: succeeded" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Operation succeeded!" service=broker_client
time="2020-04-08T08:34:04Z" level=info msg="Fetching the Runtime's dashboard URL" service=broker_client
time="2020-04-08T08:34:05Z" level=info msg="Successfully fetched dashboard URL: https://console.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com" service=broker_client
time="2020-04-08T08:34:05Z" level=info msg="Updating config map e2e-runtime-config-gcp with dashboardUrl"
time="2020-04-08T08:34:05Z" level=info msg="Fetching runtime's kubeconfig"
time="2020-04-08T08:34:05Z" level=info msg="Create request to director service" service=director_client
time="2020-04-08T08:34:05Z" level=info msg="Send request to director" service=director_client
time="2020-04-08T08:34:05Z" level=info msg="Getting authorization token for credentials to access Director from endpoint: https://oauth2.mps.dev.kyma.cloud.sap/oauth2/token"
time="2020-04-08T08:34:06Z" level=error msg="Successfully unmarshal response oauth token for accessing Director"
time="2020-04-08T08:34:06Z" level=info msg=">> variables: map[]"
time="2020-04-08T08:34:06Z" level=info msg=">> query: query {\n\tresult: runtimes(filter: { key: \"broker_instance_id\" query: \"\\\"199c898a-d2ab-4650-ad0e-6417654205ea\\\"\" }) {\n    data {\n      id\n\t}\n}\n}"
time="2020-04-08T08:34:06Z" level=info msg=">> headers: map[Accept:[application/json; charset=utf-8] Authorization:[Bearer bi_SsNePyleTRTZ5j9b_CST8t9apxjRwoXPO1ALM-kc.yYlwvaAm18iEGIaiq_rNbmzFLs0OX8gFaoeZVYBjIcQ] Content-Type:[application/json; charset=utf-8] Tenant:[3e64ebae-38b5-46a0-b1ed-9ccee153a0ae]]"
time="2020-04-08T08:34:06Z" level=info msg="<< {\"data\":{\"result\":{\"data\":[{\"id\":\"14644e6b-fbbb-4b3a-b8ac-aa97ce48f814\"}]}}}"
time="2020-04-08T08:34:06Z" level=info msg="Extract the RuntimeID from the response" service=director_client
time="2020-04-08T08:34:06Z" level=info msg="Creating a secret e2e-runtime-config-gcp with test data"
time="2020-04-08T08:34:06Z" level=info msg="Calling the dashboard URL: https://dex.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com/auth?client_id=console&nonce=79bf9c29&redirect_uri=https%3A%2F%2Fconsole.c-72f4c6b.kyma-dev.shoot.canary.k8s-hana.ondemand.com&response_type=id_token+token&scope=audience%3Aserver%3Aclient_id%3Akyma-client+audience%3Aserver%3Aclient_id%3Aconsole+openid+profile+email+groups&state=5a4f3d15" service=dashboard_checker
time="2020-04-08T08:34:06Z" level=info msg="Successful response from the dashboard URL" service=dashboard_checker
PASS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2  100     2    0     0   2000      0 --:--:-- --:--:-- --:--:--  2000
OK%
```

logs from cleanup:
```
Starting e2e-provisioning cleanup test on GCP. Waiting 20s for api server...
time="2020-04-08T09:58:45Z" level=info msg="using instance ID 9bfde7f7-8036-450d-b41e-0d812145177e"
time="2020-04-08T09:58:46Z" level=info msg="Cleaning up..."
time="2020-04-08T09:58:46Z" level=info msg="removing secret e2e-runtime-config-gcp"
time="2020-04-08T09:58:46Z" level=warning msg="secret not found"
time="2020-04-08T09:58:46Z" level=info msg="removing config map e2e-runtime-config-gcp"
time="2020-04-08T09:58:47Z" level=info msg="Create request to director service" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Send request to director" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Getting authorization token for credentials to access Director from endpoint: https://oauth2.mps.dev.kyma.cloud.sap/oauth2/token"
time="2020-04-08T09:58:47Z" level=error msg="Successfully unmarshal response oauth token for accessing Director"
time="2020-04-08T09:58:47Z" level=info msg=">> variables: map[]"
time="2020-04-08T09:58:47Z" level=info msg=">> query: query {\n\tresult: runtimes(filter: { key: \"broker_instance_id\" query: \"\\\"9bfde7f7-8036-450d-b41e-0d812145177e\\\"\" }) {\n    data {\n      id\n\t}\n}\n}"
time="2020-04-08T09:58:47Z" level=info msg=">> headers: map[Accept:[application/json; charset=utf-8] Authorization:[Bearer taXIhaNoqj8CkV14HrZpvOwKYNyvvaU9LtDNErthZWU.e0zX_sHVLS8XiBdFuu1HRSA4EjVlOZ6B2_Gdhj5mHyA] Content-Type:[application/json; charset=utf-8] Tenant:[3e64ebae-38b5-46a0-b1ed-9ccee153a0ae]]"
time="2020-04-08T09:58:47Z" level=info msg="<< {\"data\":{\"result\":{\"data\":[{\"id\":\"8bf252c1-8fa4-4836-8b9c-b11c1a00b9f9\"}]}}}"
time="2020-04-08T09:58:47Z" level=info msg="Extract the RuntimeID from the response" service=director_client
time="2020-04-08T09:58:47Z" level=info msg="Waiting for uaa-issuer instance to be removed" service=runtime_client
time="2020-04-08T09:58:48Z" level=info msg="Successfully ensured UAA instance was removed" service=runtime_client
time="2020-04-08T09:58:48Z" level=info msg="Deprovisioning Runtime [ID: 9bfde7f7-8036-450d-b41e-0d812145177e, NAME: e2e-provisioning-dtju64uqv6]" service=broker_client
time="2020-04-08T09:58:49Z" level=info msg="Successfully send deprovision request, got operation ID d852610f-857d-4f63-b41b-5255666cd78f" service=broker_client
time="2020-04-08T09:58:49Z" level=info msg="Waiting for operation at most 1h0m0s" service=broker_client
time="2020-04-08T10:03:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:08:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:13:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:18:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:23:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:28:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:33:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:38:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:43:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:48:49Z" level=info msg="Last operation status: in progress" service=broker_client
time="2020-04-08T10:53:49Z" level=info msg="Last operation status: succeeded" service=broker_client
time="2020-04-08T10:53:49Z" level=info msg="Operation succeeded!" service=broker_client
PASS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2  100     2    0     0   1000      0 --:--:-- --:--:-- --:--:--  1000
OK
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				c.log.Warn("config map not found")
+				return true, nil
+			}
+			c.log.Errorf("while creating config map: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "while waiting for secret update")
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:106]:
```suggestion
		return errors.Wrap(err, "while waiting for config map delete")
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		return err
+	}
+	return nil
+}
+
+func (c *ConfigMapClient) Update(configMap v1.ConfigMap) error {
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		err := c.client.Update(context.Background(), &configMap)
+		if err != nil {
+			c.log.Errorf("while creating config map: %v", err)
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:81]:
```suggestion
			c.log.Errorf("while updating config map: %v", err)
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	return &configMap, nil
+}
+
+func (c *ConfigMapClient) Create(configMap v1.ConfigMap) error {
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		err := c.client.Create(context.Background(), &configMap)
+		if err != nil {
+			if apiErrors.IsAlreadyExists(err) {
+				err = c.Update(configMap)
+				if err != nil {
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:61]: the polling should be finished when update is successful

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		err := c.client.Update(context.Background(), &configMap)
+		if err != nil {
+			c.log.Errorf("while creating config map: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "while waiting for secret update")
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:87]:
```suggestion
		return errors.Wrap(err, "while waiting for config map update")
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+func (c *ConfigMapClient) Delete(configMap v1.ConfigMap) error {
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		err := c.client.Delete(context.Background(), &configMap)
+		if err != nil {
+			if apiErrors.IsNotFound(err) {
+				c.log.Warn("config map not found")
+				return true, nil
+			}
+			c.log.Errorf("while creating config map: %v", err)
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:100]:
```suggestion
			c.log.Errorf("while deleting config map: %v", err)
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+					return false, errors.Wrap(err, "while updating a config map")
+				}
+			}
+			c.log.Errorf("while creating config map: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		c.log.Errorf("while creating secret: %v", err)
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:71]:
```suggestion
```

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		return err
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/configmap.go:72]: I would wrap this error and remove logging where it's not needed for consistency

File: tests/e2e/provisioning/pkg/client/v1_client/secret.go [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+				c.log.Warn("secret not found")
+				return true, nil
+			}
+			c.log.Errorf("while creating secret: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "while waiting for secret update")
```
Comment (ksputo) [inline tests/e2e/provisioning/pkg/client/v1_client/secret.go:69]: IMO errors' messages should indicate the failed method's name
```suggestion
		return errors.Wrap(err, "while waiting for secret delete")
```

File: tests/e2e/provisioning/pkg/client/runtime/runtime_client.go [hunk 9]
```diff
@@ -99,43 +117,30 @@ func (c *Client) fetchRuntimeConfig() (*runtimeStatusResponse, error) {
...
-}
-
-func (c *Client) newRuntimeClient() (client.Client, error) {
-	response, err := c.fetchRuntimeConfig()
-	if err != nil {
-		return nil, errors.Wrap(err, "while fetching runtime config")
+	if res.Result.RuntimeConfiguration != nil {
+		return res.Result.RuntimeConfiguration.Kubeconfig, nil
 	}
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply tests/e2e/provisioning/pkg/client/runtime/runtime_client.go]: done

File: tests/e2e/provisioning/pkg/client/v1_client/configmap.go (hunk 2 above)
Comment (polskikiel) [reply tests/e2e/provisioning/pkg/client/v1_client/configmap.go]: done

File: tests/e2e/provisioning/README.md [hunk 10]
```diff
@@ -16,15 +16,37 @@ You must also have the Kyma Environment Broker [configured](https://github.com/k
...
+    
+    b. Remove the test's secret and config map.
+    
+    c. Fetch the Runtime kubeconfig from Provisioner and use it to clean resources which block the cluster from deprovisioning.
+    
+    d. Send a request to deprovision the Runtime to KEB. The request is passed to Runtime Provisioner which deprovisions the Runtime.
+    
+    e. Wait until the deprovisioning is successful. It takes about 20 minutes to complete. You can configure the timeout using the environment variable.
+
+Between the above steps you can execute your own test directly on the provisioned runtime, using a kubeconfig fetched from the Provisioner.
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:49]:
```suggestion
Between the end-to-end test phases, you can execute your own test directly on the provisioned Runtime. To do so, use a kubeconfig stored in a Secret created in the provisioning phase. 
```

File: tests/e2e/provisioning/README.md [hunk 11]
```diff
@@ -47,3 +69,6 @@ You can configure the test execution by using the following environment variable
 | **APP_DIRECTOR_OAUTH_CREDENTIALS_SECRET_NAME** | Specifies the name of the Secret created by the Integration System. | `compass-kyma-environment-broker-credentials` |
 | **APP_SKIP_CERT_VERIFICATION** | Specifies whether TLS checks the presented certificates. | `false` |
 | **APP_DUMMY_TEST** | Specifies if test should success without any action. | `false` |
+| **APP_CLEANUP_PHASE** | Specifies if test is executed in the cleanup phase. | `false` |
+| **APP_CONFIG_NAME** | Specifies the name of the config map and secret, created in the test. | `false` |
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:73]:
```suggestion
| **APP_CONFIG_NAME** | Specifies the name of the ConfigMap and Secret created in the test. | `e2e-runtime-config` |
```

File: tests/e2e/provisioning/README.md [hunk 12]
```diff
@@ -47,3 +69,6 @@ You can configure the test execution by using the following environment variable
 | **APP_DIRECTOR_OAUTH_CREDENTIALS_SECRET_NAME** | Specifies the name of the Secret created by the Integration System. | `compass-kyma-environment-broker-credentials` |
 | **APP_SKIP_CERT_VERIFICATION** | Specifies whether TLS checks the presented certificates. | `false` |
 | **APP_DUMMY_TEST** | Specifies if test should success without any action. | `false` |
+| **APP_CLEANUP_PHASE** | Specifies if test is executed in the cleanup phase. | `false` |
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:72]:
```suggestion
| **APP_CLEANUP_PHASE** | Specifies if the test executes the cleanup phase. | `false` |
```

File: tests/e2e/provisioning/README.md [hunk 13]
```diff
@@ -47,3 +69,6 @@ You can configure the test execution by using the following environment variable
...
+| **APP_CONFIG_NAME** | Specifies the name of the config map and secret, created in the test. | `false` |
+| **APP_DEPLOY_NAME** | Specifies the namespace of the config map and secret, created in the test. | `false` |
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:74]:
```suggestion
| **APP_DEPLOY_NAMESPACE** | Specifies the Namespace of the ConfigMap and Secret created in the test. | `compass-system` |
```

File: tests/e2e/provisioning/README.md [hunk 14]
```diff
@@ -16,15 +16,37 @@ You must also have the Kyma Environment Broker [configured](https://github.com/k
...
+    e. Creates a Secret with a kubeconfig of the provisioned Runtime.
+    
+    f. Ensures that the DashboardURL redirects to the UUA login page. It means that the Kyma Runtime is accessible.
 
-The cleanup logic is executed at the end of the test or in case of an error. It consists of the following steps:
-1. Fetch the Runtime kubeconfig from Runtime Provisioner and use it to clean resources which block the cluster from deprovisioning.
-2. Send a request to deprovision the Runtime to KEB. The request is passed to Runtime Provisioner which deprovisions the Runtime.
+2. Cleanup
+
+    The cleanup logic is executed at the end of the end-to-end test or when the provisioning phase fails. It consists of the following steps:
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:37]:
```suggestion
    The cleanup logic is executed at the end of the end-to-end test or when the provisioning phase fails. During this phase, the test executes the following steps:
```

File: tests/e2e/provisioning/README.md [hunk 15]
```diff
@@ -16,15 +16,37 @@ You must also have the Kyma Environment Broker [configured](https://github.com/k
 
 ## Details
 
-The provisioning end-to-end test contains a broker client implementation which mocks Registry. It is an external dependency that calls the broker in the regular scenario. The test consists of the following steps:
+The provisioning end-to-end test contains a broker client implementation which mocks Registry. It is an external dependency that calls the broker in the regular scenario. The test is divided into two phases:
 
-1. Send the a call to KEB to provision a Runtime. KEB creates an operation and sends a request to Runtime Provisioner. Wait until the operation is successful. It takes about 30 minutes on GCP and a few hours on Azure. You can configure the timeout using the environment variable. 
-2. Fetch the DashboardURL from KEB.  To do so, the Runtime must be successfully provisioned and registered in the Director.
-3. Ensure that the DashboardURL redirects you to the UUA login page. It means that the Kyma Runtime is accessible.
+1. Provisioning:
```
Comment (klaudiagrz) [inline tests/e2e/provisioning/README.md:21]:
```suggestion
1. Provisioning
During the provisioning phase, the test executes the following steps:
```

**Pull request merged**